Sorting/Sorting/SortingMethods.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System.Collections.Generic;
+
+namespace Sorting
+{
+    public class SortingMethods
+    {
+        public enum Direction
+        {
+            ascending,
+            descending
+        }
+        private class Node
+        {
+            public Node Left { get; set; }
+            public Node Right { get; set; }
+            public double Value { get; set; }
+            public void Add(double value)
+            {
+                if (this.Value > value)
+                {
+                    if (this.Left == null)
+                    {
+                        this.Left = new Node { Value = value };
+                    }
+                    else
+                    {
+                        this.Left.Add(value);
+                    }
+                }
+                else
+                {
+                    if (this.Right == null)
+                    {
+                        this.Right = new Node { Value = value };
+                    }
+                    else
+                    {
+                        this.Right.Add(value);
+                    }
+                }
+            }
+        }
+
+        private static void ascendingTreeSorting(Node node, ref List<double> list)
+        {
+            if (node.Left != null)
+            {
+                ascendingTreeSorting(node.Left, ref list);
+            }
+
+            list.Add(node.Value);
+
+            if (node.Right != null)
+            {
+                ascendingTreeSorting(node.Right, ref list);
+            }
+        }
+
+        private static void descendingTreeSorting(Node node, ref List<double> list)
```

Comment:
code style problem in the name

Reply:
Understand. Thank you. My fault. It is metyhod!